xarray/namedarray/core.py

```diff
@@ -597,6 +612,54 @@ def _dask_finalize(
         data = array_func(results, *args, **kwargs)
         return type(self)(self._dims, data, attrs=self._attrs)
 
+    def load(self, **kwargs):
+        """Manually trigger loading of this variable's data from disk or a
+        remote source into memory and return this variable.
+
+        Normally, it should not be necessary to call this method in user code,
+        because all xarray functions should either work on deferred data or
+        load data automatically.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional keyword arguments passed on to ``dask.array.compute``.
+
+        See Also
+        --------
+        dask.array.compute
+        """
+        if is_chunked_array(self._data):
+            chunkmanager = get_chunked_array_type(self._data)
+            loaded_data, *_ = chunkmanager.compute(self._data, **kwargs)
+            self._data = as_compatible_data(loaded_data)
```

Comment:
Including `.load` on `NamedArray` means we would need to move `as_compatible_data` to `namedarray`, but that depends on a nuch of stuff including `duck_array_ops`. Maybe I should leave `.load` out for now @andersy005 ?

Reply:
Okay I got stuck here. I tried moving `.load`, `as_compatible_data` etc. to `xarray.namedarray.core`, but that requires `NON_NUMPY_SUPPORTED_ARRAY_TYPES` which requires `pd.Index` (as well as other pandas stuff). Presumably you don't want any pandas imports in the namedarray module, so what should I do here? 

I could import `as_compatible_data` from xarray somewhere, but it can't live in `xarray.core.variable` else I will probably have a circular import. This also wouldn't really solve the pandas dependency. Another option might be to use a less stringent coercion that `as_compatible_data` inside `NamedArray.load`.